### DISCUSSION TITLE
Better control over the deal-flow

### DISCUSSION THREAD

**Discussion by rjan90** [filecoin-project/lotus#5448] [Ideas] [answered]
Based on the discussion in this Slack-thread, I think there is a need for some features in lotus that gives the miner better control over  where the deals are stored during sealing, and how they are imported: [https://filecoinproject.slack.com/archives/G01KU7G441Y/p1611749361134800](https://filecoinproject.slack.com/archives/G01KU7G441Y/p1611749361134800).

**A couple of concerns for smaller / medium scaled miners:**

- When importing data for an offline deal, the fstmp files are placed directly in the LOTUS_MINER_PATH. Github-issue #5291.
- .lotusminer/datastore/staging getting really big when doing a lot of storage-deals
- Many smaller/medium-scaled miners are not set-up to handle larger amount of deals when deals go through the LOTUS_MINER_PATH as their disk size might not be large enough. High flow of files and IO on one of the most critical disks is not ideal.
- Computation required when importing deals. High CPU-usage is not ideal for miners that have their Lotus-node & Lotus-miner on the same machine. Miners have lost blocks due to importing deals (especially when winningPoSt is done on CPU).

**A couple of possible features to make it easier for smaller/medium scaled miners**

- Ability to set your own path to where fstmp files are stored during sealing.
- Ability to "outsource" the deal-handling to a dedicated server/pc such that the Miner doing windowPosT, winningPost, etc, is not too busy doing deals to be able to handle other more critical jobs. Example of a split-miner: [https://github.com/moran666666/lotus-1.4.0](https://github.com/moran666666/lotus-1.4.0).

**Comment by TippyFlitsUK** [comment, marked as answer]
Great suggestions above that would certainly help.

Can I also add a request for a `lotus-miner storage-deals cancel` function? 

We receive many deals that fall for numerous innocent reasons but are constantly re-tried by the miner over and over again. It would be great to have the ability to clean up our `storage-deals list` and avoid additional unnecessary miner queries.

**Reply by rjan90** [reply]
Here are two Feature Requests issues for the above discussion:
- Dedicated deal making worker: https://github.com/filecoin-project/lotus/issues/5576
- Ability to set own deal-path: https://github.com/filecoin-project/lotus/issues/5591

**Reply by rjan90** [reply]
So the last update I heard wrt to these issues was that ability to set your own deal-path will come soon (maybe one of the first updates after v1.5). While the ability to have a deal-making worker would require a lot of work, and was not going to happen for at least 3-5 month.
